gadopt/stokes_integrators.py

```diff
@@ -97,33 +97,32 @@ def create_stokes_nullspace(Z, closed=True, rotational=False, translations=None)
 class StokesSolver:
     name = 'Stokes'
 
-    def __init__(self, z, T, approximation, bcs=None, mu=1,
-                 quad_degree=6, cartesian=True, solver_parameters=None,
-                 closed=True, rotational=False, J=None, constant_jacobian=False,
-                 **kwargs):
+    def __init__(self, z, T, approximation, bcs=None, mu=1, quad_degree=6,
+                 cartesian=True, solver_parameters=None, closed=True,
+                 rotational=False, J=None, constant_jacobian=False, **kwargs):
+
         self.Z = z.function_space()
         self.mesh = self.Z.mesh()
         self.test = fd.TestFunctions(self.Z)
-        self.equations = StokesEquations(self.Z, self.Z, quad_degree=quad_degree,
```

Comment:
So based on what I see you can still inherit most of the things from `StokesSolver`. So for this here I would suggest that you keep it as it is.

Reply:
Thanks for chatting yesterday and going through this. I agree it should inherit as much as possible!